### PR TITLE
fixed a bug related to loading whole validation set

### PR DESCRIPTION
**Pull request by sadransh**
I think the current code prevents the validation set from loading completely. In fact, we are skipping the images without annotation when loading. I think this should happen when loading training images. But, when loading validation images it is better to load both with and without annotations. I saw similar behavior in the original "auto ml" code and detectron2. I added a few lines to load all images when we are loading the validation set. So, it will work similarly to automl and detectron2

### CODE REVIEW

**Comment by rwightman** [comment]
@sadransh thanks for the PR and pointing out the issue, I will leave this open for now. I have a whole whack of outstanding changes locally including a complete revamp of the datasets to support coco, voc, and openimages. I'll check the filtering criteria more closely while testing that.

**Comment by rwightman** [comment]
@sadransh 
After merging my updates the default is to include images with empty GT now. 

https://github.com/rwightman/efficientdet-pytorch/blob/master/effdet/data/parsers/parser_coco.py#L41
